datasets/openslr/openslr.py

```diff
@@ -239,18 +304,36 @@ def _generate_examples(self, path_to_indexs, path_to_datas):
         """ Yields examples. """
 
         counter = -1
-        for i, path in enumerate(path_to_indexs):
-            with open(path, encoding="utf-8") as f:
-                lines = f.readlines()
-                for id_, line in enumerate(lines):
-                    # Following regexs are needed to normalise the lines, since the datasets
-                    # are not always consistent and have bugs:
-                    line = re.sub(r"\t[^\t]*\t", "\t", line.strip())
-                    field_values = re.split(r"\t\t?", line)
-                    if len(field_values) != 2:
+        if self.config.name in ["SLR35", "SLR36"]:
+            sentence_index = {}
+            for i, path_to_index in enumerate(path_to_indexs):
+                with open(path_to_index, encoding="utf-8") as f:
+                    lines = f.readlines()
+                    for id_, line in enumerate(lines):
+                        field_values = re.split(r"\t\t?", line.strip())
+                        filename, user_id, sentence = field_values
+                        sentence_index[filename] = sentence
+                for path_to_data in sorted(Path(path_to_datas[i]).rglob("*.flac")):
+                    filename = path_to_data.stem
+                    if path_to_data.stem not in sentence_index:
                         continue
```

Comment:
Instead of running glob, do you think it would be possible to simply get the path to the audio file this way ?

```python
path = os.path.join(path_to_datas[i], f"{filename}.flac")
```

Reply:
The directory structure is more complicated. There are 16 TGZ files, each is around 1.2GB and 11700 sound files. 
Each TGZ file contains the same index file utt_spk_text.tsv for all 185076 sound files in all TGZ files. 
After un-compressing, the directory structure looks like following:

```
asr_javanese
├── data
│   ├── 00
│   │   ├── 00004fe6aa.flac
│   │   ├── 00005fb7fb.flac
│   │   ├── 0000e5df79.flac
│   ├── 01
│   │   ├── 0100691368.flac
│   │   ├── 0100f065f9.flac
│   │   ├── 01014e443a.flac
...
│   └── 0f
│       ├── 0f008f0f29.flac
│       ├── 0f00f9b7b4.flac
│       └── 0fffee991d.flac
├── LICENSE
└── utt_spk_text.tsv
```

Therefore, I use glob to get the path to the sound files.

Reply:
Ok I see ! that makes sense